src/Avalonia.Themes.Fluent/Controls/CalendarDatePicker.xaml

```diff
@@ -18,131 +11,172 @@
   </Design.PreviewWith>
 
   <Styles.Resources>
-    <sys:Double x:Key="CalendarDatePickerCurrentDayFontSize">12</sys:Double>
+    <x:Double x:Key="CalendarDatePickerCurrentDayFontSize">12</x:Double>
+    <x:Double x:Key="CalendarDatePickerMinHeight">32</x:Double>
   </Styles.Resources>
   
   <Style Selector="CalendarDatePicker">
-
     <Setter Property="Background" Value="{DynamicResource CalendarDatePickerBackground}"/>
     <Setter Property="Foreground" Value="{DynamicResource CalendarDatePickerForeground}"/>
     <Setter Property="BorderBrush" Value="{DynamicResource CalendarDatePickerBorderBrush}"/>
     <Setter Property="BorderThickness" Value="{DynamicResource CalendarDatePickerBorderThemeThickness}"/>
     <Setter Property="CornerRadius" Value="{DynamicResource ControlCornerRadius}" />
+    <Setter Property="MinHeight" Value="{DynamicResource CalendarDatePickerMinHeight}" />
     <Setter Property="HorizontalAlignment" Value="Left" />
     <Setter Property="VerticalAlignment" Value="Center" />
+    <Setter Property="HorizontalContentAlignment" Value="Left" />
+    <Setter Property="VerticalContentAlignment" Value="Center" />
     <Setter Property="Padding" Value="4"/>
-    
     <Setter Property="Template">
       <ControlTemplate>
         <DataValidationErrors>
-        <Grid ColumnDefinitions="*,Auto">
+          <Panel x:Name="LayoutRoot"
+                 HorizontalAlignment="Stretch"
+                 VerticalAlignment="Stretch">
+            <Border x:Name="Background"
+                    CornerRadius="{TemplateBinding CornerRadius}"
+                    BorderThickness="{TemplateBinding BorderThickness}"
+                    BorderBrush="{TemplateBinding BorderBrush}"
+                    Background="{TemplateBinding Background}" />
+            <Grid ColumnDefinitions="*,Auto">
+              <TextBox Name="PART_TextBox"
+                       Foreground="{TemplateBinding Foreground}"
+                       Background="Transparent"
+                       BorderBrush="Transparent"
+                       BorderThickness="0"
+                       CornerRadius="{TemplateBinding CornerRadius}"
+                       Margin="{TemplateBinding BorderThickness}"
+                       Padding="{TemplateBinding Padding}"
+                       Watermark="{TemplateBinding Watermark}"
+                       UseFloatingWatermark="{TemplateBinding UseFloatingWatermark}"
+                       HorizontalAlignment="Stretch"
+                       VerticalAlignment="Stretch"
+                       VerticalContentAlignment="{TemplateBinding VerticalContentAlignment}"
+                       HorizontalContentAlignment="{TemplateBinding HorizontalContentAlignment}"
+                       Grid.Column="0" />
+              <Button Name="PART_Button"
+                      Grid.Column="1"
+                      Focusable="False" />
+              <Popup Name="PART_Popup"
+                     PlacementTarget="{TemplateBinding}"
```

Comment:
Unexpected, didn't know this can work. But seems it does!

Reply:
That code wasn't from me -- I'm very surprised it works too. I think I'll add the path just to be clear.

Reply:
I didn't read this properly the first time. Obviously, it's similar to the `{Binding}` syntax which is perfectly valid. `{Binding}` just means bind to the current data context. `{TemplateBinding}` is similar and just means bind to the TemplateBinding context which is the parent (`TemplateBinding` is equivalent to `{Binding RelativeSource={RelativeSource TemplatedParent}`). This seems like a decent way of doing things and should just be documented somewhere. So I'm not going to change anything although I could probably replace it with the relative source equivalent binding.

Note that the Popup automatically finds a good target if one isn't set. So setting this isn't even required.

Note: I'm going into this level of detail more to document this for myself later.